conduit/telemetry/telemetry.go

```diff
@@ -0,0 +1,84 @@
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/opensearch-project/opensearch-go"
+	"github.com/opensearch-project/opensearch-go/opensearchapi"
+
+	"github.com/algorand/conduit/version"
+)
+
+// MakeTelemetryConfig initializes a new TelemetryConfig.
+func MakeTelemetryConfig(telemetryURI, index, username, password string) Config {
+	return Config{
+		Enable:   true,
+		URI:      telemetryURI,
+		GUID:     uuid.New().String(), // Use Google UUID instead of go-algorand utils
+		Index:    index,
+		UserName: username,
+		Password: password,
+	}
+}
+
+// initializeOpenSearchClient creates a new OpenSearch client.
+func initializeOpenSearchClient(cfg Config) (*opensearch.Client, error) {
+	client, err := opensearch.NewClient(opensearch.Config{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+		Addresses: []string{cfg.URI},
+		// These credentials are here intentionally. Not a bug.
+		Username: cfg.UserName,
```

Comment:
Not for this PR, but worth a discussion. We might want to leverage a library such as https://github.com/zalando/go-keyring to that users can have a way to interact with secrets stored more securely on their system than directly in the config file.

Reply:
It's a good point - we also do this in go-algorand: https://github.com/algorand/go-algorand/blob/master/logging/telemetryConfig.go#L37

We hardcode the telemetry credentials because it's write-only anyway. It's certainly weird if a user has to do this for their custom telemetry client though. We also discussed using API tokens which might be a little better instead of hardcoded passwords as well.